### ISSUE TITLE
Sending messages in Rails controller

### ISSUE THREAD

**Issue by ohararyan** [hyperoslo/facebook-messenger#13]
Am trying to trigger a message within a rails controller and not having any luck. Is this possible?

```ruby
class OrdersController < ApplicationController
  include Facebook::Messenger

  def checkout
    @order = Order.find(params[:id])
    result = Braintree::Transaction.sale()

    if result.success?
      Bot.deliver(
        recipient: {
          id: '12345'
         },
         message: {
           text: 'Human?'
         }
       )
     end
  end
end
```

If I add the following line to my controller `logger.info Facebook::Messenger::Configuration.access_token`  I can see that it is loading the config and returning the access token I have set up. 

So seems to be something odd with `Bot.deliver`. I've also tried `Facebook::Messenger::Bot.deliver` and no success either. 

When I test in the console it also returns nil

```
2.2.1 :050 > Facebook::Messenger::Bot.deliver(recipient: { id: '12345' }, message: { text: "Hello" })
 => nil 
```

**Comment by gauravtiwari** [comment]
Are you using correct recipient id @ohararyan ? Trying to reproduce this in a new Rails app now ...

**Comment by jgorset** [comment]
Hey @ohararyan! You should string describing the message ID (like `mid.1456970487936:c34767dfe57ee6e339`) back from `Bot.deliver`.

You're getting `nil` instead because the recipient you're trying to send the message to doesn't exist. Facebook is actually returning an error in the response, but we're not handling that very well. I'll change it to raise an exception, which I think would be more appropriate behaviour than returning `nil`.

You can read more about the [Send API in Facebook's documentation](https://developers.facebook.com/docs/messenger-platform/send-api-reference).

**Comment by jgorset** [comment]
tl;dr: Bots can't send messages to users that haven't sent a message to you first. The easiest way to get a valid recipient is to listen for incoming messages. You can reply straight away (like we're doing in the examples in the README) and/or save the recipient for later.

**Comment by jgorset** [comment]
It turns out I already meant for exceptions to be raised, but Facebook mucks it up by sending JSON with content type `application/javascript` instead of `application/json`. I fixed that in 67c416555 and released a patch version 0.4.1. If you update you should get a helpful exception instead of `nil`.
